projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/vdk.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import subprocess
+
+
+class VDK:
+    """
+    A single parent class containing all the individual VDK methods in it.
+    """
+
+    @staticmethod
+    def run_job(path, arguments=None):
+        """
+        Execute `run job`.
+        When no auth is provided, disables prompts for the password to avoid the terminal hanging.
+        When auth is provided, await prompts for username/passwords and sends them
+        :param path: the directory where the job run will be performed
+        :param arguments: the additional variables for the job run
+        :return: response with status code.
+        """
+        if not os.path.exists(path):
+            path = os.getcwd() + path
+            if not os.path.exists(path):
+                return "Incorrect path!"
+        p = (
+            subprocess.run(
+                f"vdk run {path} 2> vdk_logs.txt",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+            if not arguments
+            else subprocess.run(
+                f"vdk run {path} --arguments {arguments} 2> vdk_logs.txt",
```

Comment:
I looked at the VEP, and cannot find any information about vdk-core being a prerequisite. I found it rather describes that the UI is an alternative for the CLI, and it seems that actually the UI is transparently using vdk-core at least. We need to describe this to the VEP, and consider some prerequisite checks upon plugin installation(?) or another life cycle phase

Reply:
It needs to be rephrased. As the UI is alternative for the user but it still uses the CLI. I will add the changes with the Error handling section.